src/features/views/components/ViewColumnDialog/PersonFieldConfig.tsx

```diff
@@ -42,14 +41,8 @@ const PersonFieldConfig = ({
   const messages = useMessages(messageIds);
   const globalMessages = useMessages(globalMessageIds);
   const [selectedFields, setSelectedFields] = useState<string[]>([]);
-
-  //TODO: refactor to use model logic
-  const { orgId } = useRouter().query;
-  const fieldsQuery = useQuery(
-    ['customFields', orgId],
-    getCustomFields(orgId as string)
-  );
-  const customFields = fieldsQuery.data || [];
+  const { orgId } = useNumericRouteParams();
+  const customFields = useCustomFields(orgId);
```

Comment:
Seeing this in context makes me wonder why this was being done as part of the Smart Search PR, as it's a `views` feature rather than related to Smart Search. I went back and look at @kaulfield23's views PR and realized the column dialog was not dealt with there.

This indicates to me that difference in perspective that we've touched upon a couple of times. The best approach to a refactor like this is to work from the perspective of the _users_ of the code we want to change, and the users of that code are the components.

Here we have view-related components using code from other features, and I can see how it's easy to miss that when the "todo list" is the model/API modules in a feature, rather than the components.

What do I mean by "todo list" in this context? I think that it's a common thing to want to track one's progress somehow, to know when you're finished, and there are several ways to do that. One way (let's call it method A) is to look for "bad" files in the file tree, and work until they're all gone. I think that's the approach that has been used the most in this refactor. Folders like `api`, `fetching`, `models` and `repos` are indicators of badness that should be eliminated. Once they're eliminated, the work is considered done. So the programmer looks for those folders (and the files in them) within the feature they're working on, and work until they've removed all references of that code.

The other method (let's call it method B) is to look for _references_ to bad code within the feature you're refactoring, e.g. by searching within the feature folder for keywords like "react-query", "useQuery", "useModel", "queryClient" etc. That list becomes the todo list and the coder works until that list is empty. This is slightly more difficult because it requires the coder to know about everything that might be an indication of something bad.

Neither method is perfect, and a combination of both is probably ideal. But method B is definitely more focused on the _components_, i.e. the users of the code, because that's where those keywords would be found.

For features like views and Smart Search, that rely heavily on other features, using method A is bound to miss a lot of stuff, because the fetching/model code _within the feature itself_ is only a small portion of all the fetching/model code it uses, since it also uses code that lives within other features.